packages/inferno/src/DOM/mounting.ts

```diff
@@ -189,7 +188,15 @@ export function mountFunctionalComponent(
   lifecycle,
   animations: AnimationQueues
 ): void {
-  mount((vNode.children = normalizeRoot(renderFunctionalComponent(vNode, context))), parentDOM, context, isSVG, nextNode, lifecycle, animations);
+  const ref = vNode.ref;
+  // If we have a componentDidAppear on this component, we shouldn't allow children to animate so we're passing an dummy animations queue
```

Comment:
why "we shouldn't allow children to animate" ?

Reply:
If you mount a page, you could have lots of child animations that would trigger and it would kill performance really quickly. It could also cause unexpected reflow of content which would cause visual chaos.

If someone wants children to animate they can do so using the outer most animation hooks (or just through CSS based on those classes). If they know how to do it, there is a greater chance they are aware of the tradeoffs.